contracts/connectors/loantoken/LoanTokenLogicProxy.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity 0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "./AdvancedTokenStorage.sol";
+import "../../openzeppelin/Initializable.sol";
+
+/**
+ * @title Loan Token Logic Proxy contract.
+ *
+ * @notice This contract contains the proxy functionality and it will query the logic target from LoanTokenLogicBeacon
+ * This contract will also has the pause/unpause functionality. The purpose of this pausability is so that we can pause/unpause from the loan token level.
+ *
+ */
+contract LoanTokenLogicProxy is AdvancedTokenStorage {
+	/**
+	 * @notice PLEASE DO NOT ADD ANY VARIABLES HERE UNLESS FOR SPESIFIC SLOT
+	 */
+
+	/// ------------- MUST BE THE SAME AS IN LoanToken CONTRACT -------------------
+	address public sovrynContractAddress;
+	address public wrbtcTokenAddress;
+	address public target_;
+	address public admin;
+	/// ------------- END MUST BE THE SAME AS IN LoanToken CONTRACT -------------------
+
+	/**
+	 * @notice PLEASE DO NOT ADD ANY VARIABLES HERE UNLESS FOR SPESIFIC SLOT (CONSTANT / IMMUTABLE)
+	 */
+
+	bytes32 internal constant LOAN_TOKEN_LOGIC_BEACON_ADDRESS_SLOT = 0xd918085b6ac26c71bca17b569f873038d2b9c0a3a62611d7037f46a40829de6a; // keccak256("LOAN_TOKEN_LOGIC_BEACON_ADDRESS_SLOT")
+
+	modifier onlyAdmin() {
+		require(isOwner(), "LoanTokenLogicProxy:unauthorized");
+		_;
+	}
+
+	function initializeLoanTokenProxy(address _beaconAddress) public onlyAdmin {
+		// Initialize the LoanTokenLogicBeacon address
+		_setBeaconAddress(_beaconAddress);
+	}
+
+	/**
+	 * @notice Fallback function performs a logic implementation address query to LoanTokenLogicBeacon and then do delegate call to that query result address.
+	 * Returns whatever the implementation call returns.
+	 * */
+	function() external payable {
+		// query the logic target implementation address from the LoanTokenLogicBeacon
+		address target = ILoanTokenLogicBeacon(_beaconAddress()).getTarget(msg.sig);
```

Comment:
this is an issue. using the msg.sig here has the effect that no contract can interact with the loan tokens anymore. take the limit order implementation as an example. the order is executed on the settlement contract. let's assume the function is called `executeMarginOder` and it calls `marginTradeBySig`. msg.sig is then `executeMarginOder`  and the fallback here will revrt because it doesn't recognize the signature.